File: src/main/java/org/codetome/hexameter/core/api/HexagonOrientation.java
package org.codetome.hexameter.core.api;

/**
 * Enum representing the possible orientations of a {@link Hexagon}. The names
 * are self-explanatory.
 */
public enum HexagonOrientation {

    POINTY_TOP(0.5f), FLAT_TOP(0);

    private float coordinateOffset;

    HexagonOrientation(final float coordinateOffset) {
        this.coordinateOffset = coordinateOffset;
    }

    /**
     * This is because the flat/pointy shape of a hexagon.
     * It needs to be offset for pointy when calculating
     * the coordinates of its points.
     *
     * @return offset
     */
    public final float getCoordinateOffset() {
        return coordinateOffset;
    }

}

File: mixite2.core/src/commonMain/kotlin/org/hexworks/mixite2/core/api/HexagonOrientation.kt
package org.hexworks.mixite2.core.api

/**
 * Enum representing the possible orientations of a [Hexagon]. The names
 * are self-explanatory.
 */
enum class HexagonOrientation(val coordinateOffset: Float) {

    POINTY_TOP(0.5f),
    FLAT_TOP(0f)

}
